api/api/admin/media_report.py

```diff
@@ -441,69 +442,79 @@ def decision_create_view(self, request, object_id):
         reports referencing the media with this decision.
         """
 
-        if request.method == "POST":
-            media_obj = self.get_object(request, object_id)
+        redir = redirect(f"admin:api_{self.media_type}_change", object_id)
 
-            through_model = {
-                "image": ImageDecisionThrough,
-                "audio": AudioDecisionThrough,
-            }[self.media_type]
-            form = get_decision_form(self.media_type)(request.POST)
-            if form.is_valid():
-                decision = form.save(commit=False)
-                decision.moderator = request.user
-                decision.save()
-
-                logger.info(
-                    "Decision created",
-                    decision=decision.id,
-                    action=decision.action,
-                    notes=decision.notes,
-                    moderator=request.user.get_username(),
-                )
+        if request.method != "POST":
+            return redir
 
-                through = through_model.objects.create(
-                    decision=decision,
-                    media_obj=media_obj,
-                )
-                logger.info(
-                    "Through model created",
-                    through=through.id,
-                    decision=decision.id,
-                    media_obj=media_obj.id,
-                )
+        if not request.user.has_perm(f"api.add_{self.media_type}decision"):
```

Comment:
@sarayourfriend your review was accurate. I was operating under a limited understanding of Django's permission model.

I have updated the PR to be more idiomatic and use `user.has_perm` instead of checking for `is_superuser` or membership to a specific group.

Reply:
Sounds good, glad that was easy to clarify.

One more question: shouldn't this check be in the `has_add_permission` method of the model admin?

https://docs.djangoproject.com/en/5.0/ref/contrib/admin/#django.contrib.admin.ModelAdmin.has_add_permission

I'm actually surprised it isn't automatically handled as part of the model admin permissions :thinking:. I thought these two aspects of the base ModelAdmin would automatically handle these checks:

https://github.com/django/django/blob/0e49a8c3bd9119795525d9f076f73740741479b7/django/contrib/admin/options.py#L1844-L1845

and 

https://github.com/django/django/blob/0e49a8c3bd9119795525d9f076f73740741479b7/django/contrib/admin/options.py#L1836-L1837

If we're manually implementing the checks, we should have unit tests for them. Ideally we can rely on the base class behaviour... 

At the very least I think this line can be changed to:

```suggestion
        if not self.has_add_permission(request):
```

Reply:
This view (`decision_create_view`) exists in the `MediaListAdmin` class which is the admin-model class for media models (i.e. `Image` and `Audio`) so the `has_add_permission` in that class refers to the user's permission to add a new media item, and not a new decision, which is the intent of this statement.